src/targets/maven.ts

```diff
@@ -258,12 +261,91 @@ export class MavenTarget extends BaseTarget {
    * @param distDir directory of the distribution.
    */
   private async uploadDistribution(distDir: string): Promise<void> {
+    const bomFile = await this.getBomFileInDist(distDir);
+    if (bomFile) {
+      this.logger.debug('Found BOM: ', bomFile);
+      await this.uploadBomDistribution(bomFile);
+    } else {
+      this.logger.debug('Did not find a BOM.');
+      await this.uploadPomDistribution(distDir);
+    }
+  }
+
+  /**
+   * Returns the path to the BOM file in the given distribution directory, and
+   * `undefined` if there isn't any.
+   */
+  private async getBomFileInDist(distDir: string): Promise<string | undefined> {
+    const pomFilepath = join(distDir, POM_DEFAULT_FILENAME);
```

Comment:
Should this be something like `possibleBomFilePath`?

Reply:
It could, but not sure whether it _should_. We don't know whether it's a BOM, but we do know it's a POM (a BOM is a type of POM). So I think making a reference to BOM makes it less accurate; besides, some logic below treats any `xml` file as a potential BOM, so that would be confusing.

Reply:
Got it, that reasoning sounds good to me.